// @aws-sdk/client-neptune/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { NeptuneServiceException as __BaseException } from "./NeptuneServiceException";
var DBClusterNotFoundFault = (function (_super) {
    __extends(DBClusterNotFoundFault, _super);
    function DBClusterNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterNotFoundFault.prototype);
        return _this;
    }
    return DBClusterNotFoundFault;
}(__BaseException));
export { DBClusterNotFoundFault };
var DBClusterRoleAlreadyExistsFault = (function (_super) {
    __extends(DBClusterRoleAlreadyExistsFault, _super);
    function DBClusterRoleAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterRoleAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterRoleAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterRoleAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterRoleAlreadyExistsFault;
}(__BaseException));
export { DBClusterRoleAlreadyExistsFault };
var DBClusterRoleQuotaExceededFault = (function (_super) {
    __extends(DBClusterRoleQuotaExceededFault, _super);
    function DBClusterRoleQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterRoleQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterRoleQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterRoleQuotaExceededFault.prototype);
        return _this;
    }
    return DBClusterRoleQuotaExceededFault;
}(__BaseException));
export { DBClusterRoleQuotaExceededFault };
var InvalidDBClusterStateFault = (function (_super) {
    __extends(InvalidDBClusterStateFault, _super);
    function InvalidDBClusterStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBClusterStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBClusterStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBClusterStateFault.prototype);
        return _this;
    }
    return InvalidDBClusterStateFault;
}(__BaseException));
export { InvalidDBClusterStateFault };
var SourceNotFoundFault = (function (_super) {
    __extends(SourceNotFoundFault, _super);
    function SourceNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SourceNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SourceNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SourceNotFoundFault.prototype);
        return _this;
    }
    return SourceNotFoundFault;
}(__BaseException));
export { SourceNotFoundFault };
var SubscriptionNotFoundFault = (function (_super) {
    __extends(SubscriptionNotFoundFault, _super);
    function SubscriptionNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SubscriptionNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SubscriptionNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SubscriptionNotFoundFault.prototype);
        return _this;
    }
    return SubscriptionNotFoundFault;
}(__BaseException));
export { SubscriptionNotFoundFault };
var DBInstanceNotFoundFault = (function (_super) {
    __extends(DBInstanceNotFoundFault, _super);
    function DBInstanceNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceNotFoundFault.prototype);
        return _this;
    }
    return DBInstanceNotFoundFault;
}(__BaseException));
export { DBInstanceNotFoundFault };
var DBSnapshotNotFoundFault = (function (_super) {
    __extends(DBSnapshotNotFoundFault, _super);
    function DBSnapshotNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSnapshotNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSnapshotNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSnapshotNotFoundFault.prototype);
        return _this;
    }
    return DBSnapshotNotFoundFault;
}(__BaseException));
export { DBSnapshotNotFoundFault };
var ResourceNotFoundFault = (function (_super) {
    __extends(ResourceNotFoundFault, _super);
    function ResourceNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "ResourceNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceNotFoundFault.prototype);
        return _this;
    }
    return ResourceNotFoundFault;
}(__BaseException));
export { ResourceNotFoundFault };
var DBParameterGroupAlreadyExistsFault = (function (_super) {
    __extends(DBParameterGroupAlreadyExistsFault, _super);
    function DBParameterGroupAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBParameterGroupAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBParameterGroupAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBParameterGroupAlreadyExistsFault.prototype);
        return _this;
    }
    return DBParameterGroupAlreadyExistsFault;
}(__BaseException));
export { DBParameterGroupAlreadyExistsFault };
var DBParameterGroupNotFoundFault = (function (_super) {
    __extends(DBParameterGroupNotFoundFault, _super);
    function DBParameterGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBParameterGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBParameterGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBParameterGroupNotFoundFault.prototype);
        return _this;
    }
    return DBParameterGroupNotFoundFault;
}(__BaseException));
export { DBParameterGroupNotFoundFault };
var DBParameterGroupQuotaExceededFault = (function (_super) {
    __extends(DBParameterGroupQuotaExceededFault, _super);
    function DBParameterGroupQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBParameterGroupQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBParameterGroupQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBParameterGroupQuotaExceededFault.prototype);
        return _this;
    }
    return DBParameterGroupQuotaExceededFault;
}(__BaseException));
export { DBParameterGroupQuotaExceededFault };
var DBClusterSnapshotAlreadyExistsFault = (function (_super) {
    __extends(DBClusterSnapshotAlreadyExistsFault, _super);
    function DBClusterSnapshotAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterSnapshotAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterSnapshotAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterSnapshotAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterSnapshotAlreadyExistsFault;
}(__BaseException));
export { DBClusterSnapshotAlreadyExistsFault };
var DBClusterSnapshotNotFoundFault = (function (_super) {
    __extends(DBClusterSnapshotNotFoundFault, _super);
    function DBClusterSnapshotNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterSnapshotNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterSnapshotNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterSnapshotNotFoundFault.prototype);
        return _this;
    }
    return DBClusterSnapshotNotFoundFault;
}(__BaseException));
export { DBClusterSnapshotNotFoundFault };
var InvalidDBClusterSnapshotStateFault = (function (_super) {
    __extends(InvalidDBClusterSnapshotStateFault, _super);
    function InvalidDBClusterSnapshotStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBClusterSnapshotStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBClusterSnapshotStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBClusterSnapshotStateFault.prototype);
        return _this;
    }
    return InvalidDBClusterSnapshotStateFault;
}(__BaseException));
export { InvalidDBClusterSnapshotStateFault };
var KMSKeyNotAccessibleFault = (function (_super) {
    __extends(KMSKeyNotAccessibleFault, _super);
    function KMSKeyNotAccessibleFault(opts) {
        var _this = _super.call(this, __assign({ name: "KMSKeyNotAccessibleFault", $fault: "client" }, opts)) || this;
        _this.name = "KMSKeyNotAccessibleFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, KMSKeyNotAccessibleFault.prototype);
        return _this;
    }
    return KMSKeyNotAccessibleFault;
}(__BaseException));
export { KMSKeyNotAccessibleFault };
var SnapshotQuotaExceededFault = (function (_super) {
    __extends(SnapshotQuotaExceededFault, _super);
    function SnapshotQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "SnapshotQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "SnapshotQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SnapshotQuotaExceededFault.prototype);
        return _this;
    }
    return SnapshotQuotaExceededFault;
}(__BaseException));
export { SnapshotQuotaExceededFault };
var DBClusterAlreadyExistsFault = (function (_super) {
    __extends(DBClusterAlreadyExistsFault, _super);
    function DBClusterAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterAlreadyExistsFault;
}(__BaseException));
export { DBClusterAlreadyExistsFault };
var DBClusterParameterGroupNotFoundFault = (function (_super) {
    __extends(DBClusterParameterGroupNotFoundFault, _super);
    function DBClusterParameterGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterParameterGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterParameterGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterParameterGroupNotFoundFault.prototype);
        return _this;
    }
    return DBClusterParameterGroupNotFoundFault;
}(__BaseException));
export { DBClusterParameterGroupNotFoundFault };
var DBClusterQuotaExceededFault = (function (_super) {
    __extends(DBClusterQuotaExceededFault, _super);
    function DBClusterQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterQuotaExceededFault.prototype);
        return _this;
    }
    return DBClusterQuotaExceededFault;
}(__BaseException));
export { DBClusterQuotaExceededFault };
var DBSubnetGroupDoesNotCoverEnoughAZs = (function (_super) {
    __extends(DBSubnetGroupDoesNotCoverEnoughAZs, _super);
    function DBSubnetGroupDoesNotCoverEnoughAZs(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupDoesNotCoverEnoughAZs", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupDoesNotCoverEnoughAZs";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupDoesNotCoverEnoughAZs.prototype);
        return _this;
    }
    return DBSubnetGroupDoesNotCoverEnoughAZs;
}(__BaseException));
export { DBSubnetGroupDoesNotCoverEnoughAZs };
var DBSubnetGroupNotFoundFault = (function (_super) {
    __extends(DBSubnetGroupNotFoundFault, _super);
    function DBSubnetGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupNotFoundFault.prototype);
        return _this;
    }
    return DBSubnetGroupNotFoundFault;
}(__BaseException));
export { DBSubnetGroupNotFoundFault };
var GlobalClusterNotFoundFault = (function (_super) {
    __extends(GlobalClusterNotFoundFault, _super);
    function GlobalClusterNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "GlobalClusterNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "GlobalClusterNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, GlobalClusterNotFoundFault.prototype);
        return _this;
    }
    return GlobalClusterNotFoundFault;
}(__BaseException));
export { GlobalClusterNotFoundFault };
var InsufficientStorageClusterCapacityFault = (function (_super) {
    __extends(InsufficientStorageClusterCapacityFault, _super);
    function InsufficientStorageClusterCapacityFault(opts) {
        var _this = _super.call(this, __assign({ name: "InsufficientStorageClusterCapacityFault", $fault: "client" }, opts)) || this;
        _this.name = "InsufficientStorageClusterCapacityFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InsufficientStorageClusterCapacityFault.prototype);
        return _this;
    }
    return InsufficientStorageClusterCapacityFault;
}(__BaseException));
export { InsufficientStorageClusterCapacityFault };
var InvalidDBInstanceStateFault = (function (_super) {
    __extends(InvalidDBInstanceStateFault, _super);
    function InvalidDBInstanceStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBInstanceStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBInstanceStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBInstanceStateFault.prototype);
        return _this;
    }
    return InvalidDBInstanceStateFault;
}(__BaseException));
export { InvalidDBInstanceStateFault };
var InvalidDBSubnetGroupStateFault = (function (_super) {
    __extends(InvalidDBSubnetGroupStateFault, _super);
    function InvalidDBSubnetGroupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSubnetGroupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSubnetGroupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSubnetGroupStateFault.prototype);
        return _this;
    }
    return InvalidDBSubnetGroupStateFault;
}(__BaseException));
export { InvalidDBSubnetGroupStateFault };
var InvalidGlobalClusterStateFault = (function (_super) {
    __extends(InvalidGlobalClusterStateFault, _super);
    function InvalidGlobalClusterStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidGlobalClusterStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidGlobalClusterStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidGlobalClusterStateFault.prototype);
        return _this;
    }
    return InvalidGlobalClusterStateFault;
}(__BaseException));
export { InvalidGlobalClusterStateFault };
var InvalidSubnet = (function (_super) {
    __extends(InvalidSubnet, _super);
    function InvalidSubnet(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidSubnet", $fault: "client" }, opts)) || this;
        _this.name = "InvalidSubnet";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidSubnet.prototype);
        return _this;
    }
    return InvalidSubnet;
}(__BaseException));
export { InvalidSubnet };
var InvalidVPCNetworkStateFault = (function (_super) {
    __extends(InvalidVPCNetworkStateFault, _super);
    function InvalidVPCNetworkStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidVPCNetworkStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidVPCNetworkStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidVPCNetworkStateFault.prototype);
        return _this;
    }
    return InvalidVPCNetworkStateFault;
}(__BaseException));
export { InvalidVPCNetworkStateFault };
var StorageQuotaExceededFault = (function (_super) {
    __extends(StorageQuotaExceededFault, _super);
    function StorageQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "StorageQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "StorageQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, StorageQuotaExceededFault.prototype);
        return _this;
    }
    return StorageQuotaExceededFault;
}(__BaseException));
export { StorageQuotaExceededFault };
var DBClusterEndpointAlreadyExistsFault = (function (_super) {
    __extends(DBClusterEndpointAlreadyExistsFault, _super);
    function DBClusterEndpointAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterEndpointAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterEndpointAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterEndpointAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterEndpointAlreadyExistsFault;
}(__BaseException));
export { DBClusterEndpointAlreadyExistsFault };
var DBClusterEndpointQuotaExceededFault = (function (_super) {
    __extends(DBClusterEndpointQuotaExceededFault, _super);
    function DBClusterEndpointQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterEndpointQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterEndpointQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterEndpointQuotaExceededFault.prototype);
        return _this;
    }
    return DBClusterEndpointQuotaExceededFault;
}(__BaseException));
export { DBClusterEndpointQuotaExceededFault };
var AuthorizationNotFoundFault = (function (_super) {
    __extends(AuthorizationNotFoundFault, _super);
    function AuthorizationNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "AuthorizationNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "AuthorizationNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AuthorizationNotFoundFault.prototype);
        return _this;
    }
    return AuthorizationNotFoundFault;
}(__BaseException));
export { AuthorizationNotFoundFault };
var DBInstanceAlreadyExistsFault = (function (_super) {
    __extends(DBInstanceAlreadyExistsFault, _super);
    function DBInstanceAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceAlreadyExistsFault.prototype);
        return _this;
    }
    return DBInstanceAlreadyExistsFault;
}(__BaseException));
export { DBInstanceAlreadyExistsFault };
var DBSecurityGroupNotFoundFault = (function (_super) {
    __extends(DBSecurityGroupNotFoundFault, _super);
    function DBSecurityGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSecurityGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSecurityGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSecurityGroupNotFoundFault.prototype);
        return _this;
    }
    return DBSecurityGroupNotFoundFault;
}(__BaseException));
export { DBSecurityGroupNotFoundFault };
var DomainNotFoundFault = (function (_super) {
    __extends(DomainNotFoundFault, _super);
    function DomainNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DomainNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DomainNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DomainNotFoundFault.prototype);
        return _this;
    }
    return DomainNotFoundFault;
}(__BaseException));
export { DomainNotFoundFault };
var InstanceQuotaExceededFault = (function (_super) {
    __extends(InstanceQuotaExceededFault, _super);
    function InstanceQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "InstanceQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "InstanceQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InstanceQuotaExceededFault.prototype);
        return _this;
    }
    return InstanceQuotaExceededFault;
}(__BaseException));
export { InstanceQuotaExceededFault };
var InsufficientDBInstanceCapacityFault = (function (_super) {
    __extends(InsufficientDBInstanceCapacityFault, _super);
    function InsufficientDBInstanceCapacityFault(opts) {
        var _this = _super.call(this, __assign({ name: "InsufficientDBInstanceCapacityFault", $fault: "client" }, opts)) || this;
        _this.name = "InsufficientDBInstanceCapacityFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InsufficientDBInstanceCapacityFault.prototype);
        return _this;
    }
    return InsufficientDBInstanceCapacityFault;
}(__BaseException));
export { InsufficientDBInstanceCapacityFault };
var OptionGroupNotFoundFault = (function (_super) {
    __extends(OptionGroupNotFoundFault, _super);
    function OptionGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "OptionGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "OptionGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, OptionGroupNotFoundFault.prototype);
        return _this;
    }
    return OptionGroupNotFoundFault;
}(__BaseException));
export { OptionGroupNotFoundFault };
var ProvisionedIopsNotAvailableInAZFault = (function (_super) {
    __extends(ProvisionedIopsNotAvailableInAZFault, _super);
    function ProvisionedIopsNotAvailableInAZFault(opts) {
        var _this = _super.call(this, __assign({ name: "ProvisionedIopsNotAvailableInAZFault", $fault: "client" }, opts)) || this;
        _this.name = "ProvisionedIopsNotAvailableInAZFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ProvisionedIopsNotAvailableInAZFault.prototype);
        return _this;
    }
    return ProvisionedIopsNotAvailableInAZFault;
}(__BaseException));
export { ProvisionedIopsNotAvailableInAZFault };
var StorageTypeNotSupportedFault = (function (_super) {
    __extends(StorageTypeNotSupportedFault, _super);
    function StorageTypeNotSupportedFault(opts) {
        var _this = _super.call(this, __assign({ name: "StorageTypeNotSupportedFault", $fault: "client" }, opts)) || this;
        _this.name = "StorageTypeNotSupportedFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, StorageTypeNotSupportedFault.prototype);
        return _this;
    }
    return StorageTypeNotSupportedFault;
}(__BaseException));
export { StorageTypeNotSupportedFault };
var DBSubnetGroupAlreadyExistsFault = (function (_super) {
    __extends(DBSubnetGroupAlreadyExistsFault, _super);
    function DBSubnetGroupAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupAlreadyExistsFault.prototype);
        return _this;
    }
    return DBSubnetGroupAlreadyExistsFault;
}(__BaseException));
export { DBSubnetGroupAlreadyExistsFault };
var DBSubnetGroupQuotaExceededFault = (function (_super) {
    __extends(DBSubnetGroupQuotaExceededFault, _super);
    function DBSubnetGroupQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupQuotaExceededFault.prototype);
        return _this;
    }
    return DBSubnetGroupQuotaExceededFault;
}(__BaseException));
export { DBSubnetGroupQuotaExceededFault };
var DBSubnetQuotaExceededFault = (function (_super) {
    __extends(DBSubnetQuotaExceededFault, _super);
    function DBSubnetQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetQuotaExceededFault.prototype);
        return _this;
    }
    return DBSubnetQuotaExceededFault;
}(__BaseException));
export { DBSubnetQuotaExceededFault };
var EventSubscriptionQuotaExceededFault = (function (_super) {
    __extends(EventSubscriptionQuotaExceededFault, _super);
    function EventSubscriptionQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "EventSubscriptionQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "EventSubscriptionQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, EventSubscriptionQuotaExceededFault.prototype);
        return _this;
    }
    return EventSubscriptionQuotaExceededFault;
}(__BaseException));
export { EventSubscriptionQuotaExceededFault };
var SNSInvalidTopicFault = (function (_super) {
    __extends(SNSInvalidTopicFault, _super);
    function SNSInvalidTopicFault(opts) {
        var _this = _super.call(this, __assign({ name: "SNSInvalidTopicFault", $fault: "client" }, opts)) || this;
        _this.name = "SNSInvalidTopicFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SNSInvalidTopicFault.prototype);
        return _this;
    }
    return SNSInvalidTopicFault;
}(__BaseException));
export { SNSInvalidTopicFault };
var SNSNoAuthorizationFault = (function (_super) {
    __extends(SNSNoAuthorizationFault, _super);
    function SNSNoAuthorizationFault(opts) {
        var _this = _super.call(this, __assign({ name: "SNSNoAuthorizationFault", $fault: "client" }, opts)) || this;
        _this.name = "SNSNoAuthorizationFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SNSNoAuthorizationFault.prototype);
        return _this;
    }
    return SNSNoAuthorizationFault;
}(__BaseException));
export { SNSNoAuthorizationFault };
var SNSTopicArnNotFoundFault = (function (_super) {
    __extends(SNSTopicArnNotFoundFault, _super);
    function SNSTopicArnNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SNSTopicArnNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SNSTopicArnNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SNSTopicArnNotFoundFault.prototype);
        return _this;
    }
    return SNSTopicArnNotFoundFault;
}(__BaseException));
export { SNSTopicArnNotFoundFault };
var SubscriptionAlreadyExistFault = (function (_super) {
    __extends(SubscriptionAlreadyExistFault, _super);
    function SubscriptionAlreadyExistFault(opts) {
        var _this = _super.call(this, __assign({ name: "SubscriptionAlreadyExistFault", $fault: "client" }, opts)) || this;
        _this.name = "SubscriptionAlreadyExistFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SubscriptionAlreadyExistFault.prototype);
        return _this;
    }
    return SubscriptionAlreadyExistFault;
}(__BaseException));
export { SubscriptionAlreadyExistFault };
var SubscriptionCategoryNotFoundFault = (function (_super) {
    __extends(SubscriptionCategoryNotFoundFault, _super);
    function SubscriptionCategoryNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SubscriptionCategoryNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SubscriptionCategoryNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SubscriptionCategoryNotFoundFault.prototype);
        return _this;
    }
    return SubscriptionCategoryNotFoundFault;
}(__BaseException));
export { SubscriptionCategoryNotFoundFault };
var GlobalClusterAlreadyExistsFault = (function (_super) {
    __extends(GlobalClusterAlreadyExistsFault, _super);
    function GlobalClusterAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "GlobalClusterAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "GlobalClusterAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, GlobalClusterAlreadyExistsFault.prototype);
        return _this;
    }
    return GlobalClusterAlreadyExistsFault;
}(__BaseException));
export { GlobalClusterAlreadyExistsFault };
var GlobalClusterQuotaExceededFault = (function (_super) {
    __extends(GlobalClusterQuotaExceededFault, _super);
    function GlobalClusterQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "GlobalClusterQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "GlobalClusterQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, GlobalClusterQuotaExceededFault.prototype);
        return _this;
    }
    return GlobalClusterQuotaExceededFault;
}(__BaseException));
export { GlobalClusterQuotaExceededFault };
var DBClusterEndpointNotFoundFault = (function (_super) {
    __extends(DBClusterEndpointNotFoundFault, _super);
    function DBClusterEndpointNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterEndpointNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterEndpointNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterEndpointNotFoundFault.prototype);
        return _this;
    }
    return DBClusterEndpointNotFoundFault;
}(__BaseException));
export { DBClusterEndpointNotFoundFault };
var InvalidDBClusterEndpointStateFault = (function (_super) {
    __extends(InvalidDBClusterEndpointStateFault, _super);
    function InvalidDBClusterEndpointStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBClusterEndpointStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBClusterEndpointStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBClusterEndpointStateFault.prototype);
        return _this;
    }
    return InvalidDBClusterEndpointStateFault;
}(__BaseException));
export { InvalidDBClusterEndpointStateFault };
var InvalidDBParameterGroupStateFault = (function (_super) {
    __extends(InvalidDBParameterGroupStateFault, _super);
    function InvalidDBParameterGroupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBParameterGroupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBParameterGroupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBParameterGroupStateFault.prototype);
        return _this;
    }
    return InvalidDBParameterGroupStateFault;
}(__BaseException));
export { InvalidDBParameterGroupStateFault };
var DBSnapshotAlreadyExistsFault = (function (_super) {
    __extends(DBSnapshotAlreadyExistsFault, _super);
    function DBSnapshotAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSnapshotAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSnapshotAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSnapshotAlreadyExistsFault.prototype);
        return _this;
    }
    return DBSnapshotAlreadyExistsFault;
}(__BaseException));
export { DBSnapshotAlreadyExistsFault };
var InvalidDBSubnetStateFault = (function (_super) {
    __extends(InvalidDBSubnetStateFault, _super);
    function InvalidDBSubnetStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSubnetStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSubnetStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSubnetStateFault.prototype);
        return _this;
    }
    return InvalidDBSubnetStateFault;
}(__BaseException));
export { InvalidDBSubnetStateFault };
var InvalidEventSubscriptionStateFault = (function (_super) {
    __extends(InvalidEventSubscriptionStateFault, _super);
    function InvalidEventSubscriptionStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidEventSubscriptionStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidEventSubscriptionStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidEventSubscriptionStateFault.prototype);
        return _this;
    }
    return InvalidEventSubscriptionStateFault;
}(__BaseException));
export { InvalidEventSubscriptionStateFault };
var InvalidDBSecurityGroupStateFault = (function (_super) {
    __extends(InvalidDBSecurityGroupStateFault, _super);
    function InvalidDBSecurityGroupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSecurityGroupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSecurityGroupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSecurityGroupStateFault.prototype);
        return _this;
    }
    return InvalidDBSecurityGroupStateFault;
}(__BaseException));
export { InvalidDBSecurityGroupStateFault };
var SharedSnapshotQuotaExceededFault = (function (_super) {
    __extends(SharedSnapshotQuotaExceededFault, _super);
    function SharedSnapshotQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "SharedSnapshotQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "SharedSnapshotQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SharedSnapshotQuotaExceededFault.prototype);
        return _this;
    }
    return SharedSnapshotQuotaExceededFault;
}(__BaseException));
export { SharedSnapshotQuotaExceededFault };
var CertificateNotFoundFault = (function (_super) {
    __extends(CertificateNotFoundFault, _super);
    function CertificateNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "CertificateNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "CertificateNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CertificateNotFoundFault.prototype);
        return _this;
    }
    return CertificateNotFoundFault;
}(__BaseException));
export { CertificateNotFoundFault };
var DBUpgradeDependencyFailureFault = (function (_super) {
    __extends(DBUpgradeDependencyFailureFault, _super);
    function DBUpgradeDependencyFailureFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBUpgradeDependencyFailureFault", $fault: "client" }, opts)) || this;
        _this.name = "DBUpgradeDependencyFailureFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBUpgradeDependencyFailureFault.prototype);
        return _this;
    }
    return DBUpgradeDependencyFailureFault;
}(__BaseException));
export { DBUpgradeDependencyFailureFault };
var SubnetAlreadyInUse = (function (_super) {
    __extends(SubnetAlreadyInUse, _super);
    function SubnetAlreadyInUse(opts) {
        var _this = _super.call(this, __assign({ name: "SubnetAlreadyInUse", $fault: "client" }, opts)) || this;
        _this.name = "SubnetAlreadyInUse";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SubnetAlreadyInUse.prototype);
        return _this;
    }
    return SubnetAlreadyInUse;
}(__BaseException));
export { SubnetAlreadyInUse };
var DBClusterRoleNotFoundFault = (function (_super) {
    __extends(DBClusterRoleNotFoundFault, _super);
    function DBClusterRoleNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterRoleNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterRoleNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterRoleNotFoundFault.prototype);
        return _this;
    }
    return DBClusterRoleNotFoundFault;
}(__BaseException));
export { DBClusterRoleNotFoundFault };
var InsufficientDBClusterCapacityFault = (function (_super) {
    __extends(InsufficientDBClusterCapacityFault, _super);
    function InsufficientDBClusterCapacityFault(opts) {
        var _this = _super.call(this, __assign({ name: "InsufficientDBClusterCapacityFault", $fault: "client" }, opts)) || this;
        _this.name = "InsufficientDBClusterCapacityFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InsufficientDBClusterCapacityFault.prototype);
        return _this;
    }
    return InsufficientDBClusterCapacityFault;
}(__BaseException));
export { InsufficientDBClusterCapacityFault };
var InvalidDBSnapshotStateFault = (function (_super) {
    __extends(InvalidDBSnapshotStateFault, _super);
    function InvalidDBSnapshotStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSnapshotStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSnapshotStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSnapshotStateFault.prototype);
        return _this;
    }
    return InvalidDBSnapshotStateFault;
}(__BaseException));
export { InvalidDBSnapshotStateFault };
var InvalidRestoreFault = (function (_super) {
    __extends(InvalidRestoreFault, _super);
    function InvalidRestoreFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidRestoreFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidRestoreFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidRestoreFault.prototype);
        return _this;
    }
    return InvalidRestoreFault;
}(__BaseException));
export { InvalidRestoreFault };
export var AddRoleToDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AddSourceIdentifierToSubscriptionMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventSubscriptionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AddSourceIdentifierToSubscriptionResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AddTagsToResourceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ApplyPendingMaintenanceActionMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PendingMaintenanceActionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ResourcePendingMaintenanceActionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ApplyPendingMaintenanceActionResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CopyDBClusterParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterParameterGroupFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CopyDBClusterParameterGroupResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CopyDBClusterSnapshotMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterSnapshotFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CopyDBClusterSnapshotResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CopyDBParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBParameterGroupFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CopyDBParameterGroupResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterRoleFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterMemberFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterOptionGroupStatusFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var VpcSecurityGroupMembershipFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterEndpointMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterEndpointOutputFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterParameterGroupResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterSnapshotMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBClusterSnapshotResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBInstanceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBParameterGroupStatusFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBSecurityGroupMembershipFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AvailabilityZoneFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SubnetFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBSubnetGroupFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DomainMembershipFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EndpointFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var OptionGroupMembershipFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PendingCloudwatchLogsExportsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PendingModifiedValuesFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBInstanceStatusInfoFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBInstanceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBInstanceResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBParameterGroupResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBSubnetGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDBSubnetGroupResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateEventSubscriptionMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateEventSubscriptionResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateGlobalClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GlobalClusterMemberFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GlobalClusterFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateGlobalClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBClusterEndpointMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBClusterEndpointOutputFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBClusterParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBClusterSnapshotMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBClusterSnapshotResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBInstanceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBInstanceResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDBSubnetGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEventSubscriptionMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEventSubscriptionResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteGlobalClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteGlobalClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterEndpointFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterEndpointMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FilterFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBClusterEndpointsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterParameterGroupsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBClusterParameterGroupsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ParameterFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterParameterGroupDetailsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBClusterParametersMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBClustersMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBClusterSnapshotAttributesMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterSnapshotAttributeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterSnapshotAttributesResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBClusterSnapshotAttributesResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterSnapshotMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBClusterSnapshotsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CharacterSetFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TimezoneFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpgradeTargetFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBEngineVersionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBEngineVersionMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBEngineVersionsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBInstanceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBInstancesMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBParameterGroupsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBParameterGroupsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBParameterGroupDetailsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBParametersMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBSubnetGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeDBSubnetGroupsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeEngineDefaultClusterParametersMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EngineDefaultsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeEngineDefaultClusterParametersResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeEngineDefaultParametersMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeEngineDefaultParametersResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeEventCategoriesMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventCategoriesMapFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventCategoriesMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeEventsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeEventSubscriptionsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventSubscriptionsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeGlobalClustersMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GlobalClustersMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeOrderableDBInstanceOptionsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var OrderableDBInstanceOptionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var OrderableDBInstanceOptionsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribePendingMaintenanceActionsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PendingMaintenanceActionsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeValidDBInstanceModificationsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DoubleRangeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RangeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ValidStorageOptionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ValidDBInstanceModificationsMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeValidDBInstanceModificationsResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FailoverDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FailoverDBClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FailoverGlobalClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FailoverGlobalClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTagsForResourceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagListMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CloudwatchLogsExportConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBClusterEndpointMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBClusterEndpointOutputFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBClusterParameterGroupNameMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBClusterParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBClusterSnapshotAttributeMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBClusterSnapshotAttributeResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBInstanceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBInstanceResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DBParameterGroupNameMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBSubnetGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyDBSubnetGroupResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyEventSubscriptionMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyEventSubscriptionResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyGlobalClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ModifyGlobalClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PromoteReadReplicaDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PromoteReadReplicaDBClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RebootDBInstanceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RebootDBInstanceResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RemoveFromGlobalClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RemoveFromGlobalClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RemoveRoleFromDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RemoveSourceIdentifierFromSubscriptionMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RemoveSourceIdentifierFromSubscriptionResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RemoveTagsFromResourceMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ResetDBClusterParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ResetDBParameterGroupMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RestoreDBClusterFromSnapshotMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RestoreDBClusterFromSnapshotResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RestoreDBClusterToPointInTimeMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RestoreDBClusterToPointInTimeResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StartDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StartDBClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StopDBClusterMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StopDBClusterResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
