// @aws-sdk/client-neptune/dist-es/commands/ModifyGlobalClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyGlobalClusterMessageFilterSensitiveLog, ModifyGlobalClusterResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryModifyGlobalClusterCommand, serializeAws_queryModifyGlobalClusterCommand, } from "../protocols/Aws_query";
var ModifyGlobalClusterCommand = (function (_super) {
    __extends(ModifyGlobalClusterCommand, _super);
    function ModifyGlobalClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyGlobalClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "NeptuneClient";
        var commandName = "ModifyGlobalClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyGlobalClusterMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ModifyGlobalClusterResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyGlobalClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyGlobalClusterCommand(input, context);
    };
    ModifyGlobalClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyGlobalClusterCommand(output, context);
    };
    return ModifyGlobalClusterCommand;
}($Command));
export { ModifyGlobalClusterCommand };
