// @aws-sdk/client-neptune/dist-es/commands/ModifyDBClusterSnapshotAttributeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBClusterSnapshotAttributeMessageFilterSensitiveLog, ModifyDBClusterSnapshotAttributeResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryModifyDBClusterSnapshotAttributeCommand, serializeAws_queryModifyDBClusterSnapshotAttributeCommand, } from "../protocols/Aws_query";
var ModifyDBClusterSnapshotAttributeCommand = (function (_super) {
    __extends(ModifyDBClusterSnapshotAttributeCommand, _super);
    function ModifyDBClusterSnapshotAttributeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBClusterSnapshotAttributeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "NeptuneClient";
        var commandName = "ModifyDBClusterSnapshotAttributeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBClusterSnapshotAttributeMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBClusterSnapshotAttributeResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBClusterSnapshotAttributeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBClusterSnapshotAttributeCommand(input, context);
    };
    ModifyDBClusterSnapshotAttributeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBClusterSnapshotAttributeCommand(output, context);
    };
    return ModifyDBClusterSnapshotAttributeCommand;
}($Command));
export { ModifyDBClusterSnapshotAttributeCommand };
