// @aws-sdk/client-neptune/dist-es/commands/RestoreDBClusterToPointInTimeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RestoreDBClusterToPointInTimeMessageFilterSensitiveLog, RestoreDBClusterToPointInTimeResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryRestoreDBClusterToPointInTimeCommand, serializeAws_queryRestoreDBClusterToPointInTimeCommand, } from "../protocols/Aws_query";
var RestoreDBClusterToPointInTimeCommand = (function (_super) {
    __extends(RestoreDBClusterToPointInTimeCommand, _super);
    function RestoreDBClusterToPointInTimeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RestoreDBClusterToPointInTimeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "NeptuneClient";
        var commandName = "RestoreDBClusterToPointInTimeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RestoreDBClusterToPointInTimeMessageFilterSensitiveLog,
            outputFilterSensitiveLog: RestoreDBClusterToPointInTimeResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RestoreDBClusterToPointInTimeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRestoreDBClusterToPointInTimeCommand(input, context);
    };
    RestoreDBClusterToPointInTimeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRestoreDBClusterToPointInTimeCommand(output, context);
    };
    return RestoreDBClusterToPointInTimeCommand;
}($Command));
export { RestoreDBClusterToPointInTimeCommand };
